File: iosApp/iosApp/Presentation/Main/CheckIn/Map/MapUiState.swift
//
//  MapUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 4/23/26.
//

import Foundation
import Shared

struct MapUiState {
    var isLoading: Bool = false
    var errorMessage: String? = nil
    var isLoggedIn: Bool = false
    var isLoginPromptVisible: Bool = false
    var currentLocationLabel: String = ""
    var userCityKey: String? = nil
    var mapCafes: [CheckInCafeSummary] = []
    var selectedRegion: ExploreUiState.RegionFilter = .all
    var searchQuery: String = ""

    static let empty = MapUiState()
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/checkin/map/MapUiState.kt
package com.hhp227.concafe.presentation.main.checkin.map

import com.hhp227.concafe.domain.model.CheckInCafeSummary
import com.hhp227.concafe.presentation.main.explore.ExploreUiState

data class MapUiState(
    val isLoading: Boolean = false,
    val errorMessage: String? = null,
    val currentLocationLabel: String = "",
    val userCityKey: String? = null,
    val mapCafes: List<CheckInCafeSummary> = emptyList(),
    val selectedRegion: ExploreUiState.RegionFilter = ExploreUiState.RegionFilter.ALL,
    val searchQuery: String = ""
) {
    companion object {
        fun empty() = MapUiState()
    }
}
